fivetran_provider_async/hooks.py

```diff
@@ -616,57 +670,68 @@ async def get_sync_status_async(
+        :param propagate_failures_forward: If True, then this method will always
+            raise an AirflowException when the most recent connector status is
+            a failure and there are no successes dated after the target time.
+            Specifically, this makes it so that
+            `completed_after_time > failed_at > succeeded_at` is considered a
+            fail condition.
         """
         connector_details = await self.get_connector_async(connector_id)
-        succeeded_at = self._parse_timestamp(connector_details["succeeded_at"])
-        failed_at = self._parse_timestamp(connector_details["failed_at"])
-        current_completed_at = succeeded_at if succeeded_at > failed_at else failed_at
-
-        # The only way to tell if a sync failed is to check if its latest
-        # failed_at value is greater than then last known "sync completed at" value.
-        if failed_at > previous_completed_at:
-            service_name = connector_details["service"]
-            schema_name = connector_details["schema"]
-            raise AirflowException(
-                f"Fivetran sync for connector {connector_id} failed; "
-                f"please see logs at "
-                f"{self._connector_ui_url_logs(service_name, schema_name)}"
-            )
-
-        sync_state = connector_details["status"]["sync_state"]
-        self.log.info('Connector "%s": sync_state = "%s"', connector_id, sync_state)
-
-        # if sync in rescheduled start, wait for time recommended by Fivetran
-        # or manually specified, then restart sync
-        if sync_state == "rescheduled" and connector_details["schedule_type"] == "manual":
-            self.log.info('Connector is in "rescheduled" state and needs to be manually restarted')
-            self.pause_and_restart(
-                connector_id=connector_id,
-                reschedule_for=connector_details["status"]["rescheduled_for"],
-                reschedule_wait_time=reschedule_wait_time,
-            )
-
-        # Check if sync started by airflow has finished
-        # indicated by new 'succeeded_at' timestamp
-        if current_completed_at > previous_completed_at:
-            self.log.info(
-                'Connector "%s": succeeded_at = "%s"', connector_id, succeeded_at.to_iso8601_string()
-            )
-            job_status = "success"
-            return job_status
+        is_completed = self._determine_if_synced_from_connector_details(
+            connector_id=connector_id,
+            connector_details=connector_details,
+            completed_after_time=completed_after_time,
+            reschedule_wait_time=reschedule_wait_time,
+            always_wait_when_syncing=always_wait_when_syncing,
+            propagate_failures_forward=propagate_failures_forward,
+        )
+        if is_completed:
+            return "success"
         else:
-            job_status = "pending"
-            return job_status
+            return "pending"
```

Comment:
```suggestion
        return "pending"
```

I think we can remove this else block

Reply:
+1